### DISCUSSION TITLE
About reverse proxy config with caddy2

### DISCUSSION THREAD

**Discussion by moonrailgun** [socketio/socket.io#4379] [Q&A]
In old version of socketio 3.x, i just still can look up caddy config. (https://socket.io/docs/v3/reverse-proxy/)

And in socketio 4.x, caddy is disappear. Is it means socketio 4.x not support caddy? 

Because of i am try to access socket.io server behind proxy with cross-origin, ws request always timeout. And if i dont use caddy, everything will be fine

**Comment by darrachequesne** [comment]
Hmm this was indeed lost during the migration from v3 to v4, but Caddy is still supported. I have updated the website accordingly: https://socket.io/docs/v4/reverse-proxy/#caddy-2

Could you please provide the `Caddyfile` configuration that you are using?

**Reply by moonrailgun** [reply]
Oh, thanks of your reply, after i found all issues / google / forum of caddy and cannot fix this problem. i switch to nginx and everything is working.

this is my final caddyfile:
```bash
domain:443 {
  encode gzip
  reverse_proxy * http://127.0.0.1:13001 {
     header_up Host {upstream_hostport}
  }
}
```
